scripts/west_commands/runners/mdb.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2020 Synopsys.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for Synopsys Metaware Debugger(mdb).'''
+
+
+from os import path
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+class MdbBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for mdb.'''
+
+    def __init__(self, cfg, cores=1, jtag='digilent', nsim_args='', gui=True,
+                dig_device=''):
+        super(MdbBinaryRunner, self).__init__(cfg)
+        self.jtag = jtag
+        self.cores = cores
+        if nsim_args != '':
+            self.nsim_args = path.join(cfg.board_dir, 'support', nsim_args)
+        else:
+            self.nsim_args = ''
+        self.gui = gui
+        self.elf_name = cfg.elf_file
+        self.dig_device = dig_device
+        self.commander = 'mdb'
+
+    @classmethod
+    def name(cls):
+        return 'mdb'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash', 'debug'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--jtag', default='digilent',
+                            help='choose the jtag interface')
+        parser.add_argument('--cores', default=1,
+                            help='choose the cores to debug')
+        parser.add_argument('--gui', default=True,
+                            help='enable gui debug')
+        parser.add_argument('--nsim_args', default='',
+                            help='if given, arguments for nsim simulator')
+        parser.add_argument('--dig-device', default='',
+                            help='''choose the digilent device when
+                             --jtag=digilent''')
+
+    @classmethod
+    def create(cls, cfg, args):
+        return MdbBinaryRunner(
+            cfg,
+            cores=args.cores,
+            jtag=args.jtag,
+            nsim_args=args.nsim_args,
+            gui=args.gui,
+            dig_device=args.dig_device
+            )
+
+    def do_run(self, command, **kwargs):
```

Comment:
We may also want to implement `do_debug`, right?

Reply:
do_debug is handled in
```
  if command == 'flash':
            mdb_run = ['-run', '-cl']
        else:
            if self.gui:
                mdb_run = ['-OKN']
            else:
                mdb_run = ['-cl']

```

Reply:
OK I thought `do_run()` & `do_debug()` are called by upstream script but they are in fact only used in the runner itself, so then it's OK.